### PR TITLE
Private/gokay/guides

### PR DESCRIPTION
**Pull request by gokaysatir**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by gokaysatir** [comment]
Snapping to positions will be added with a final commit.

File: browser/src/app/GraphicSelectionMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+class GraphicSelection {
+    public static rectangle: cool.SimpleRectangle = null;
+    public static extraInfo: any = null;
+	public static selectionAngle: number = 0;
```
Comment (eszkadev) [inline browser/src/app/GraphicSelectionMiddleware.ts:16]: white space is not unified here and there

File: browser/.beforeprettier [hunk 2]
```diff
@@ -93,6 +93,7 @@
 /src/core/Util.js
 /src/docstate.js
 /src/app/DocEvents.ts
+/src/app/GraphicSelectionMiddleware.ts
```
Comment (eszkadev) [inline browser/.beforeprettier:96]: please DO NOT add new code to the breforeprettier
this is only for OLD code we don't want to format

File: browser/src/canvas/sections/ShapeHandlesSection.ts [hunk 3]
```diff
@@ -617,9 +641,87 @@ class ShapeHandlesSection extends CanvasSectionObject {
...
-			this.sendTransformCommand(point);
+		if (this.containerObject.isDraggingSomething()) {
+			if ((window as any).mode.isTablet() || (window as any).mode.isMobile())
+				this.sendTransformCommand(point);
+			else if ((window as any).mode.isDesktop() && (this.sectionProperties.closestX || this.sectionProperties.closestY)) {
+				// We need to snap to the guide-lines. So we send a positioning command after the mouse up event (desktop case).
+				// We don't need to do this on mobile because we always send the positioning commands there. No mouse events for mobile.
+				setTimeout(() => {
+					this.sendTransformCommand(point); // Send to back of the process queue so it performs after buttonup event is sent.
+				}, 10);
```
Comment (eszkadev) [inline browser/src/canvas/sections/ShapeHandlesSection.ts:652]: in that case I prefer "0" so it is added to the queue immediately without risk of something between

File: browser/src/control/Control.FormulaBarJSDialog.js [hunk 4]
```diff
@@ -45,7 +45,6 @@ class FormulaBar {
 		this.map.off('jsdialogaction', this.onJSAction, this);
 
 		this.map.off('doclayerinit', this.onDocLayerInit, this);
-		this.map.off('updatepermission', this.onUpdatePermission, this);
```
Comment (eszkadev) [inline browser/src/control/Control.FormulaBarJSDialog.js:48]: we should have `app.events.off`

**Comment by eszkadev** [comment]
Please rebase, solve conflict, remove beforeprettier for new file and run formatting.
It's new thing, I would wait for the next week's RED phase.

File: browser/src/control/Control.FormulaBarJSDialog.js (hunk 4 above)
Comment (gokaysatir) [reply browser/src/control/Control.FormulaBarJSDialog.js:48]: When is this function called? I checked it now and couldn't trigger it. Maybe wen can remove this function.

Comment (eszkadev) [reply browser/src/control/Control.FormulaBarJSDialog.js:48]: I think it is triggered when we switch edit -> readonly mode for example, easy to check on tablet, or when disconnected the server or connected again.

Comment (gokaysatir) [reply browser/src/control/Control.FormulaBarJSDialog.js:48]: I couldn't hit the codepath unfortunately.

Comment (eszkadev) [reply browser/src/control/Control.FormulaBarJSDialog.js:48]: please see `browser/src/control/Permission.js` `_enterEditMode` and `_enterReadOnlyMode`

Comment (gokaysatir) [reply browser/src/control/Control.FormulaBarJSDialog.js:48]: I can't hit there with/without an integrator on mobile view. Let me check this later again.

**Review by eszkadev — APPROVED** [review]

**Pull request merged**